Test panic

## Bug Report

### Description

https://ci.dev.talos-systems.io/talos-systems/talos/9137/1/29

### Logs

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x93cd0c]
goroutine 557 [running]:
github.com/talos-systems/talos/pkg/client.(*Client).ServiceInfo(0xc000a00b90, 0x1a703e0, 0xc0010c26f0, 0x1813fca, 0x8, 0x0, 0x0, 0x0, 0xc000bc6c60, 0x40c0dc, ...)
	/src/pkg/client/client.go:599 +0x10c
github.com/talos-systems/talos/internal/pkg/cluster/check.ServiceStateAssertion(0x1a703a0, 0xc001387380, 0x1a77120, 0xc0001de460, 0x1813fca, 0x8, 0xc0014efa80, 0x2, 0x2, 0x26428e0, ...)
	/src/internal/pkg/cluster/check/service.go:34 +0x179
github.com/talos-systems/talos/internal/pkg/cluster/check.DefaultClusterChecks.func2.1(0x1a703a0, 0xc001387380, 0x12a05f200, 0x1a703a0)
	/src/internal/pkg/cluster/check/default.go:27 +0xc8
github.com/talos-systems/talos/internal/pkg/conditions.(*pollingCondition).Wait.func1(0x1a703a0, 0xc000acd6e0, 0xc00101f6d0, 0x0, 0x0)
	/src/internal/pkg/conditions/poll.go:55 +0x93
github.com/talos-systems/talos/internal/pkg/conditions.(*pollingCondition).Wait(0xc00101f6d0, 0x1a703a0, 0xc000acd6e0, 0x0, 0x0)
	/src/internal/pkg/conditions/poll.go:63 +0x10d
github.com/talos-systems/talos/internal/pkg/cluster/check.Wait.func1(0xc001387200, 0x1a703a0, 0xc000acd6e0, 0x1a3f660, 0xc00101f6d0)
	/src/internal/pkg/cluster/check/check.go:52 +0x45
created by github.com/talos-systems/talos/internal/pkg/cluster/check.Wait
	/src/internal/pkg/cluster/check/check.go:51 +0x127
```

### Environment

- Talos version: [`talosctl version --nodes <problematic nodes>`]
- Kubernetes version: [`kubectl version --short`]
- Platform:

--- a/pkg/client/client.go
+++ b/pkg/client/client.go
@@ -592,12 +592,18 @@ func (c *Client) ServiceInfo(ctx context.Context, id string, callOptions ...grpc
 	)
 
 	if err != nil {
-		return
+		return services, err
 	}
 
 	var filtered interface{}
 	filtered, err = FilterMessages(resp, err)
 	resp, _ = filtered.(*machineapi.ServiceListResponse) //nolint: errcheck
+
+	// FilterMessages might remove responses if they actually contain errors,
+	// errors will be merged into `resp`.
+	if resp == nil {
+		return services, err
+	}
 
 	for _, resp := range resp.Messages {
 		for _, svc := range resp.Services {
@@ -610,7 +616,7 @@ func (c *Client) ServiceInfo(ctx context.Context, id string, callOptions ...grpc
 		}
 	}
 
-	return
+	return services, err
 }
 
 // ServiceStart starts a service.

